src/circulating_supply/circulating_supply_provider.rs

```diff
@@ -0,0 +1,150 @@
+use bigdecimal::{BigDecimal, ToPrimitive};
+use diesel::PgConnection;
+
+use crate::circulating_supply::lockup::{
+    get_account_state, get_code_version, is_bug_inside, TRANSFERS_ENABLED,
+};
+use crate::circulating_supply::user_balance::get_user_balance;
+use crate::db_adapters::accounts::collect_lockups_for_block;
+use crate::db_adapters::blocks::closest_block_for;
+use crate::db_adapters::circulating_supply::{
+    add_circulating_supply, get_precomputed_circulating_supply,
+};
+use crate::models;
+use crate::models::circulating_supply::CirculatingSupply;
+use crate::models::Block;
+use actix::Addr;
+use actix_web::rt::time;
+use std::ops::Div;
+use std::str::FromStr;
+use std::time::SystemTime;
+use tracing::info;
```

Comment:
It would be ideal if we can preserve the code-style around the imports: https://github.com/near/near-indexer-for-explorer/blob/master/CONTRIBUTING.md#imports-ordering-and-grouping

It helps me to review the project structure.